src/main/java/unrc/dose/Proposition.java

```diff
@@ -1,18 +1,417 @@
+        final char[] str1,
+        final char[] str2) {
+        int[][] distance = new int[str1.length + 1][str2.length + 1];
+
+        for (int i = 0; i <= str1.length; i++) {
+            distance[i][0] = i;
+        }
+        for (int j = 0; j <= str2.length; j++) {
+            distance[0][j] = j;
+        }
+        for (int i = 1; i <= str1.length; i++) {
+            for (int j = 1; j <= str2.length; j++) {
+                distance[i][j] = minimum(distance[i - 1][j] + 1,
+                        distance[i][j - 1] + 1, distance[i - 1][j - 1]
+                                + ((str1[i - 1] == str2[j - 1]) ? 0 : 1));
+            }
+        }
+        return distance[str1.length][str2.length];
+    }
+
+    /**
+     * Generate a .java file with the code of a challenge.
+     * @param source text with code java
+     */
+    public static void generateFileJava(final String source) {
+
+        File f;
+        f = new File("Source.java");
+
+        try {
+            FileWriter w = new FileWriter(f);
+            BufferedWriter bw = new BufferedWriter(w);
+            PrintWriter wr = new PrintWriter(bw);
+            wr.write(source);
+            wr.close();
+            bw.close();
+        }
+        catch (IOException e) {
+        };
+    }
+
+    /**
+     * Compile a file .java.
+     * @param archivoSourceJava file .java
+     * @return 0 if the compilation is successful, 1 opposite case
+     * @throws Exception
+     */
+    private static int compilar(
+            final String archivoSourceJava) throws Exception {
+        int k = runProcess("javac " +  archivoSourceJava);
+        return k;
+    }
+
+    /**
+     * The code to be compiled into the buffer.
+     * @param name param javac Nombre.java
+     * @param ins
+     * @throws Exception
+     */
+    private static void printLines(
+            final String name, final InputStream ins) throws Exception {
+        String line = null;
+        BufferedReader in = new BufferedReader(new InputStreamReader(ins));
+        while ((line = in.readLine()) != null) {
+            System.out.println(name + " " + line);
+        }
+    }
+
+    /**
+     * run the compilation process.
+     * @param command java Nombre.java
+     * @return 0 if the compilation is successful, 1 opposite case
+     * @throws Exception
+     */
+    private static int runProcess(final String command) throws Exception {
+        Process pro = Runtime.getRuntime().exec(command);
+        printLines(command + " stdout:", pro.getInputStream());
+        printLines(command + " stderr:", pro.getErrorStream());
+        pro.waitFor();
+        return pro.exitValue();
+    }
+
+    /**
+     * Compile a proposal.
+     * @param proposedCode represents the code for the possible solution.
+     * @return True if compiled, false otherwise
+     */
+    public boolean compileProposition(final String proposedCode) {
+        this.setSource(proposedCode);
+        this.saveIt();
```

Comment:
why does compiling a proposition needs to change the `Proposition` state and save it to the DB?